src/primitives/transaction.h

```diff
@@ -50,6 +50,8 @@ class COutPoint
     std::string ToString() const;
 };
 
+std::size_t hash_value(const COutPoint& o);
```

Comment:
If that's the case then it's better to make less changes. We could submit other PR to improve.